src/js/components/invoice/InvoiceListFilters.jsx

```diff
@@ -1,161 +1,38 @@
 
-import { fetchInvoiceStatuses, fetchInvoiceTypeCodes, fetchSuppliers } from 'actions';
 import FilterForm from 'components/Filter/FilterForm';
-import DateFilter from 'components/form-elements/DateFilter';
-import FilterSelectField from 'components/form-elements/FilterSelectField';
+import filterFields from 'components/invoice/FilterFields';
+import useInvoiceFilters from 'hooks/useInvoiceFilters';
 import { debounceUsersFetch } from 'utils/option-utils';
 
-const filterFields = {
-  buyerOrganization: {
-    type: FilterSelectField,
-    attributes: {
-      valueKey: 'id',
-      filterElement: true,
-      placeholder: 'react.invoice.buyerOrganization.label',
-      defaultPlaceholder: 'Buyer organization',
-      showLabelTooltip: true,
-      disabled: true,
-    },
-    getDynamicAttr: ({ organization }) => ({
-      options: [
-        {
-          id: organization.id,
-          value: organization.id,
-          name: organization.name,
-          label: organization.name,
-        },
-      ],
-    }),
-  },
-  status: {
-    type: FilterSelectField,
-    attributes: {
-      filterElement: true,
-      placeholder: 'react.invoice.status.label',
-      defaultPlaceholder: 'Invoice Status',
-      showLabelTooltip: true,
-    },
-    getDynamicAttr: ({ statuses }) => ({
-      options: statuses,
-    }),
-  },
-  vendor: {
-    type: FilterSelectField,
-    attributes: {
-      valueKey: 'id',
-      filterElement: true,
-      placeholder: 'react.invoice.vendor.label',
-      defaultPlaceholder: 'Vendor',
-      showLabelTooltip: true,
-    },
-    getDynamicAttr: ({ suppliers }) => ({
-      options: suppliers,
-    }),
-  },
-  invoiceTypeCode: {
-    type: FilterSelectField,
-    attributes: {
-      filterElement: true,
-      placeholder: 'react.invoice.typeCode.label',
-      defaultPlaceholder: 'Invoice Type',
-      showLabelTooltip: true,
-    },
-    getDynamicAttr: ({ typeCodes }) => ({
-      options: typeCodes,
-    }),
-  },
-  dateInvoiced: {
-    type: DateFilter,
-    attributes: {
-      dateFormat: 'MM/DD/YYYY',
-      filterElement: true,
-      label: 'react.invoice.invoiceDate.label',
-      defaultMessage: 'Invoice Date',
-    },
-  },
-  createdBy: {
-    type: FilterSelectField,
-    attributes: {
-      async: true,
-      openOnClick: false,
-      autoload: false,
-      cache: false,
-      valueKey: 'id',
-      labelKey: 'name',
-      options: [],
-      filterOptions: options => options,
-      filterElement: true,
-      placeholder: 'react.invoice.createdBy.label',
-      defaultPlaceholder: 'Created by',
-      showLabelTooltip: true,
-    },
-    getDynamicAttr: ({
-      debouncedUsersFetch,
-    }) => ({
-      loadOptions: debouncedUsersFetch,
-    }),
-  },
-};
-
 const InvoiceListFilters = ({
   setFilterParams,
   debounceTime,
   minSearchLength,
   currentLocation,
   statuses,
-  fetchStatuses,
   suppliers,
-  fetchSupplierOrganizations,
   typeCodes,
-  fetchTypeCodes,
+  history,
```

Comment:
Do you think bumping and router-dom should be included in this ticket if we used to use history using withRouter all around the project?

Reply:
I believe as soon as you don’t need anything specific (like I needed with redux hooks), we shouldn’t bump up anything additionally. Imo they should be some small tech debt tickets where we could include some „smoke tests” to see if nothing broke

Reply:
This was just my suggestion to avoid passing `history` as a parameter and use `useHistory()` instead. I doubt that there would be much refactoring in the project since version 5 does not remove `withRouter` HOC, but we probably don't really need it and it is not that big of a problem.

Reply:
If the update is not "problematic", we could make it. To not avoid a bigger refactor in the future and replace everywhere the history props to hook. Unless there are some potential issues, then we can skip it for now.